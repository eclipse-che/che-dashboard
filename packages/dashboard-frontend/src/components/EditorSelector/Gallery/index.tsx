/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import { Gallery } from '@patternfly/react-core';
import React from 'react';

import { EditorSelectorEntry } from '@/components/EditorSelector/Gallery/Entry';
import { che } from '@/services/models';

export type Props = {
  defaultEditorId: string;
  editors: che.Plugin[];
  selectedEditorId: string | undefined;
  onSelect: (editorId: string) => void;
};
export type State = {
  selectedId: string;
  sortedEditorsByName: Map<string, che.Plugin[]>;
};

export class EditorGallery extends React.PureComponent<Props, State> {
  constructor(props: Props) {
    super(props);

    this.state = {
      selectedId: '', // will be set on component mount
      sortedEditorsByName: new Map<string, che.Plugin[]>(),
    };
  }

  public componentDidMount(): void {
    this.init();
  }

  public componentDidUpdate(prevProps: Readonly<Props>): void {
    if (prevProps.selectedEditorId !== this.props.selectedEditorId) {
      this.init();
    }
  }

  private init(): void {
    const { defaultEditorId, editors, selectedEditorId, onSelect } = this.props;

    const sortedEditors = sortEditors(editors);

    const sortedEditorsByName = new Map<string, che.Plugin[]>();

    let defaultEditor: che.Plugin | undefined;
    let selectedEditor: che.Plugin | undefined;
    sortedEditors.forEach(editor => {
      const name = editor.name;
      if (!sortedEditorsByName.has(name)) {
        sortedEditorsByName.set(name, []);
      }
      sortedEditorsByName.get(name)?.push(editor);

      // find the default editor
      if (editor.id === defaultEditorId) {
        defaultEditor = editor;
      }
      // find the selected editor
      if (editor.id === selectedEditorId) {
        selectedEditor = editor;
      }
    });

    let selectedId: string;
    if (selectedEditor !== undefined) {
      selectedId = selectedEditor.id;
    } else {
      if (defaultEditor !== undefined) {
        selectedId = defaultEditor.id;
      } else {
        selectedId = sortedEditors[0].id;
      }
      onSelect(selectedId);
    }

    this.setState({
      selectedId,
      sortedEditorsByName,
    });
  }

  private handleEditorSelect(editorId: string): void {
    this.props.onSelect(editorId);
  }

  private buildEditorCards(): React.ReactNode[] {
    const { selectedId, sortedEditorsByName } = this.state;

    return Array.from(sortedEditorsByName.keys()).map(editorName => {
      // editors same name, different version
      const editorsGroup = sortedEditorsByName.get(editorName);

      /* c8 ignore next 3 */
      if (editorsGroup === undefined) {
        return;
      }

      const groupIcon = editorsGroup[0].icon;
      const groupName = editorsGroup[0].displayName || editorsGroup[0].name;

      return (
        <EditorSelectorEntry
          key={editorName}
          editorsGroup={editorsGroup}
          groupIcon={groupIcon}
          groupName={groupName}
          selectedId={selectedId}
          onSelect={editorId => this.handleEditorSelect(editorId)}
        />
      );
    });
  }

  public render() {
    return (
      <Gallery hasGutter={true} minWidths={{ default: '170px' }} maxWidths={{ default: '170px' }}>
        {this.buildEditorCards()}
      </Gallery>
    );
  }
}

const VERSION_PRIORITY: ReadonlyArray<string> = ['insiders', 'next', 'latest'];
export function sortEditors(editors: che.Plugin[]) {
  const sorted = editors.sort((a, b) => {
    if (a.name === b.name) {
      const aPriority = VERSION_PRIORITY.indexOf(a.version);
      const bPriority = VERSION_PRIORITY.indexOf(b.version);

      if (aPriority !== -1 && bPriority !== -1) {
        return aPriority - bPriority;
      } else if (aPriority !== -1) {
        return -1;
      } else if (bPriority !== -1) {
        return 1;
      }
    }

    return a.id.localeCompare(b.id);
  });

  return sorted;
}
