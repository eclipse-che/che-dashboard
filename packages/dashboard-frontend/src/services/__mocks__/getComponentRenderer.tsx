/*
 * Copyright (c) 2018-2021 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

export default function <T extends Array<unknown>>(
  getComponent: (..._args: T) => React.ReactElement,
): {
  createSnapshot: (...args: T) => renderer.ReactTestRenderer;
  renderComponent: (...args: T) => {
    reRenderComponent: (..._args: T) => void;
  };
} {
  return {
    createSnapshot: (...args: T) => {
      return renderer.create(getComponent(...args));
    },
    renderComponent: (...args: T) => {
      const res = render(getComponent(...args));
      return {
        reRenderComponent: (..._args: T) => {
          res.rerender(getComponent(..._args));
        },
      };
    },
  };
}
