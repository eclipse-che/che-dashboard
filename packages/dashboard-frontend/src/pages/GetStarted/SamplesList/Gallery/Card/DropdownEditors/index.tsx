/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import {
  Dropdown,
  DropdownGroup,
  DropdownItem,
  DropdownPosition,
  KebabToggle,
} from '@patternfly/react-core';
import { CheckIcon } from '@patternfly/react-icons';
import React from 'react';

import TagLabel from '@/components/TagLabel';
import { PluginEditor } from '@/pages/GetStarted/SamplesList/Gallery';
import styles from '@/pages/GetStarted/SamplesList/Gallery/Card/DropdownEditors/index.module.css';

export type Props = {
  editors: PluginEditor[];
  onEditorSelect: (editorId: string) => void;
};

type State = {
  isExpanded: boolean;
};

export class DropdownEditors extends React.PureComponent<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = {
      isExpanded: false,
    };
  }

  private handleClick(editorId: string): void {
    this.props.onEditorSelect(editorId);
  }

  private toggleExpand(isExpanded: boolean): void {
    this.setState({ isExpanded });
  }

  private getDropdownItemGroup(): React.ReactNode[] {
    const { editors } = this.props;

    return [
      <DropdownGroup
        className={styles.dropdownEditorGroup}
        label="Choose an editor"
        key="editor-group"
      >
        {editors.map(editor => {
          return (
            <DropdownItem
              key={editor.id}
              tooltip={editor.displayName}
              onClick={() => this.handleClick(editor.id)}
              data-testid="card-action"
            >
              <span className={styles.editorTitle}>{editor.name}</span>
              <TagLabel version={editor.version} />
              {editor.isDefault && (
                <CheckIcon data-testid="checkmark" className={styles.checkIcon} />
              )}
            </DropdownItem>
          );
        })}
      </DropdownGroup>,
    ];
  }

  public render(): React.ReactElement {
    const { isExpanded } = this.state;

    const dropdownItems = this.getDropdownItemGroup();

    return (
      <Dropdown
        style={{ whiteSpace: 'nowrap' }}
        toggle={<KebabToggle onToggle={isExpanded => this.toggleExpand(isExpanded)} />}
        isOpen={isExpanded}
        position={DropdownPosition.right}
        dropdownItems={dropdownItems}
        isPlain
        data-testid="card-actions-dropdown"
        onClick={e => e.stopPropagation()}
      />
    );
  }
}
