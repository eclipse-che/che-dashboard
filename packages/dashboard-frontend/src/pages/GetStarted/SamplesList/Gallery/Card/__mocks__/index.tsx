/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { Props } from '@/pages/GetStarted/SamplesList/Gallery/Card';

export class SampleCard extends React.PureComponent<Props> {
  render(): React.ReactElement {
    const { metadata, onClick } = this.props;

    return (
      <div data-testid="sample-card">
        <div data-testid="sample-name">{metadata.displayName}</div>
        <button onClick={() => onClick()}>Select Sample</button>
      </div>
    );
  }
}
