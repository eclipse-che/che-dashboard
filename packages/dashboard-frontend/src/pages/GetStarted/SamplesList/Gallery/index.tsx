/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import {
  Button,
  EmptyState,
  EmptyStateBody,
  EmptyStateIcon,
  EmptyStatePrimary,
  EmptyStateVariant,
  Gallery,
  Title,
} from '@patternfly/react-core';
import { SearchIcon } from '@patternfly/react-icons';
import React from 'react';
import { connect, ConnectedProps } from 'react-redux';

import { SampleCard } from '@/pages/GetStarted/SamplesList/Gallery/Card';
import { che } from '@/services/models';
import * as DevfileRegistriesStore from '@/store/DevfileRegistries';
import { DevfileRegistryMetadata, EMPTY_WORKSPACE_TAG } from '@/store/DevfileRegistries/selectors';

export type PluginEditor = che.Plugin & {
  isDefault: boolean;
};

export const VISIBLE_TAGS = ['Community', 'Tech-Preview', 'Devfile.io'];

export type Props = MappedProps & {
  metadataFiltered: DevfileRegistryMetadata[];
  onCardClick: (metadata: DevfileRegistryMetadata) => void;
};

export class SamplesListGallery extends React.PureComponent<Props> {
  private handleCardClick(metadata: DevfileRegistryMetadata): void {
    this.props.onCardClick(metadata);
  }

  private prepareMetadata(): DevfileRegistryMetadata[] {
    function sortByVisibleTag(a: DevfileRegistryMetadata, b: DevfileRegistryMetadata): -1 | 0 | 1 {
      const getVisibleTag = (metadata: DevfileRegistryMetadata) =>
        metadata.tags.filter(tag => VISIBLE_TAGS.includes(tag))[0];
      const tagA = getVisibleTag(a);
      const tagB = getVisibleTag(b);
      if (tagA === tagB) {
        return 0;
      }
      if (tagA === undefined || tagA < tagB) {
        return -1;
      }
      if (tagB === undefined || tagA > tagB) {
        return 1;
      }
      return 0;
    }

    function sortByEmptyWorkspaceTag(
      a: DevfileRegistryMetadata,
      b: DevfileRegistryMetadata,
    ): -1 | 0 | 1 {
      if (a.tags.includes(EMPTY_WORKSPACE_TAG) > b.tags.includes(EMPTY_WORKSPACE_TAG)) {
        return -1;
      }
      if (a.tags.includes(EMPTY_WORKSPACE_TAG) < b.tags.includes(EMPTY_WORKSPACE_TAG)) {
        return 1;
      }
      return 0;
    }

    function sortByDisplayName(a: DevfileRegistryMetadata, b: DevfileRegistryMetadata): -1 | 0 | 1 {
      if (a.displayName < b.displayName) {
        return -1;
      }
      if (a.displayName > b.displayName) {
        return 1;
      }
      return 0;
    }

    return this.props.metadataFiltered
      .sort(sortByDisplayName)
      .sort(sortByVisibleTag)
      .sort(sortByEmptyWorkspaceTag);
  }

  private buildCardsList(): React.ReactElement[] {
    const metadata = this.prepareMetadata();

    return metadata.map(meta => (
      <SampleCard key={meta.links.v2} metadata={meta} onClick={() => this.handleCardClick(meta)} />
    ));
  }

  render(): React.ReactElement {
    const cards = this.buildCardsList();

    if (cards.length === 0) {
      return (
        <EmptyState variant={EmptyStateVariant.full}>
          <EmptyStateIcon icon={SearchIcon} />
          <Title headingLevel="h1">No results found</Title>
          <EmptyStateBody>
            No results match the filter criteria. Clear filter to show results.
          </EmptyStateBody>
          <EmptyStatePrimary>
            <Button variant="link" onClick={(): void => this.props.clearFilter()}>
              Clear filter
            </Button>
          </EmptyStatePrimary>
        </EmptyState>
      );
    }

    return <Gallery hasGutter={true}>{cards}</Gallery>;
  }
}

const connector = connect(null, {
  ...DevfileRegistriesStore.actionCreators,
});

type MappedProps = ConnectedProps<typeof connector>;
export default connector(SamplesListGallery);
