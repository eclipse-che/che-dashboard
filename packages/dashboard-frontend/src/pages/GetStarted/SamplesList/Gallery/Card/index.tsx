/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import {
  Badge,
  Brand,
  Card,
  CardActions,
  CardBody,
  CardHeader,
  CardHeaderMain,
} from '@patternfly/react-core';
import { CubesIcon } from '@patternfly/react-icons';
import React from 'react';

import { VISIBLE_TAGS } from '@/pages/GetStarted/SamplesList/Gallery';
import styles from '@/pages/GetStarted/SamplesList/Gallery/Card/index.module.css';
import { che } from '@/services/models';
import { convertIconToSrc } from '@/services/registry/devfiles';
import { DevfileRegistryMetadata } from '@/store/DevfileRegistries/selectors';

export type Props = {
  metadata: DevfileRegistryMetadata;
  onClick: () => void;
};

export class SampleCard extends React.PureComponent<Props> {
  private getTags(): JSX.Element[] {
    const {
      metadata: { tags },
    } = this.props;

    const createTag = (text: string, key: number): React.ReactElement => {
      return (
        <Badge
          isRead
          style={{ whiteSpace: 'nowrap' }}
          key={`badge_${key}`}
          data-testid="card-badge"
        >
          {text.trim()}
        </Badge>
      );
    };

    return tags
      .filter(tag => VISIBLE_TAGS.indexOf(tag) !== -1)
      .map((item: string, index: number) => createTag(item, index));
  }

  private handleCardClick(): void {
    this.props.onClick();
  }

  private buildIcon(metadata: che.DevfileMetaData): React.ReactElement {
    const props = {
      className: styles.sampleCardIcon,
      alt: metadata.displayName,
      'aria-label': metadata.displayName,
    };

    return metadata.icon ? (
      <Brand src={convertIconToSrc(metadata.icon)} {...props} data-testid="sample-card-icon" />
    ) : (
      <CubesIcon {...props} data-testid="default-card-icon" />
    );
  }

  render(): React.ReactElement {
    const { metadata } = this.props;

    const tags = this.getTags();
    const devfileIcon = this.buildIcon(metadata);

    return (
      <Card
        isFlat
        isCompact
        isSelectable
        key={metadata.links.v2}
        onClick={() => this.handleCardClick()}
        className={'sample-card'}
        data-testid="sample-card"
      >
        <CardHeader>
          <CardHeaderMain>{devfileIcon}</CardHeaderMain>
          <CardActions>{tags}</CardActions>
        </CardHeader>
        <CardHeader>{metadata.displayName}</CardHeader>
        <CardBody>{metadata.description}</CardBody>
      </Card>
    );
  }
}
