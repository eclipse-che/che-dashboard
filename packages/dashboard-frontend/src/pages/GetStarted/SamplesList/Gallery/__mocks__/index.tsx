/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { Props } from '@/pages/GetStarted/SamplesList/Gallery';

export default class SamplesListGallery extends React.PureComponent<Props> {
  render() {
    const { onCardClick, metadataFiltered = [] } = this.props;
    return (
      <div data-testid="samples-list-gallery">
        <div>Samples List Gallery</div>
        {metadataFiltered.map(metadata => (
          <div key={metadata.links.v2} data-testid="sample-card">
            {metadata.displayName}

            <button
              onClick={() => {
                onCardClick(metadata);
              }}
            >
              Select Sample
            </button>
          </div>
        ))}
      </div>
    );
  }
}
