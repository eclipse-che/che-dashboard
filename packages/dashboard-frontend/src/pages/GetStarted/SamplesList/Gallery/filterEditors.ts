/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import { PluginEditor } from '@/pages/GetStarted/SamplesList/Gallery';

/**
 * Filter the editors to leave the first occurrence of each editor name.
 */
export function filterMostPrioritized(editors: PluginEditor[]): PluginEditor[] {
  const editorNames = new Set();
  const filteredEditors = editors.filter(editor => {
    if (!editorNames.has(editor.name)) {
      editorNames.add(editor.name);
      return true;
    }
    return false;
  });

  return filteredEditors;
}

/**
 * Sort editors by name and version. The priority is defined as follows:
 * 1. Default editor
 * 2. Version insiders
 * 3. Version next
 * 4. Version latest
 * The rest of the editors are sorted by id.
 */
export function sortByPriority(editors: PluginEditor[]) {
  function sortFn(editorA: PluginEditor, editorB: PluginEditor) {
    const priority = ['isDefault', 'insiders', 'next', 'latest'];

    if (editorA.name === editorB.name) {
      for (const prop of priority) {
        if (editorA.version === prop) return -1;
        if (editorB.version === prop) return 1;
      }
    }
    return editorA.id.localeCompare(editorB.id);
  }

  return editors.sort(sortFn);
}
