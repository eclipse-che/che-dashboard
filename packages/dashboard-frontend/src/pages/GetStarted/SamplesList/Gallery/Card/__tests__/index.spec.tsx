/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import userEvent from '@testing-library/user-event';
import React from 'react';

import { SampleCard } from '@/pages/GetStarted/SamplesList/Gallery/Card';
import getComponentRenderer, { screen } from '@/services/__mocks__/getComponentRenderer';
import { DevfileRegistryMetadata } from '@/store/DevfileRegistries/selectors';

const onCardClick = jest.fn();

const { createSnapshot, renderComponent } = getComponentRenderer(getComponent);

describe('Devfile Metadata Card', () => {
  let metadata: DevfileRegistryMetadata;

  beforeEach(() => {
    metadata = {
      displayName: 'Go',
      description: 'Stack with Go 1.12.10',
      tags: ['Debian', 'Go'],
      icon: '/images/go.svg',
      globalMemoryLimit: '1686Mi',
      links: {
        v2: '/devfiles/go/devfile.yaml',
      },
    };
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  test('snapshot', () => {
    const snapshot = createSnapshot(metadata);
    expect(snapshot.toJSON()).toMatchSnapshot();
  });

  it('should have a correct title in header', () => {
    renderComponent(metadata);
    const cardHeader = screen.getByText(metadata.displayName);
    expect(cardHeader).toBeTruthy();
  });

  it('should have an icon', () => {
    renderComponent(metadata);
    const cardIcon = screen.queryByTestId('sample-card-icon');
    expect(cardIcon).toBeTruthy();
  });

  it('should be able to provide the default icon', () => {
    metadata.icon = '';
    renderComponent(metadata);

    const sampleIcon = screen.queryByTestId('sample-card-icon');
    expect(sampleIcon).toBeFalsy();

    const blankIcon = screen.getByTestId('default-card-icon');
    expect(blankIcon).toBeTruthy();
  });

  it('should not have visible tags', () => {
    metadata.tags = ['Debian', 'Go'];
    renderComponent(metadata);

    const badge = screen.queryAllByTestId('card-badge');
    expect(badge.length).toEqual(0);
  });

  it('should have "Community" tag', () => {
    metadata.tags = ['Community', 'Debian', 'Go'];
    renderComponent(metadata);

    const badge = screen.queryAllByTestId('card-badge');
    expect(badge.length).toEqual(1);
    expect(screen.queryByText('Community')).toBeTruthy();
  });

  it('should have "tech-preview" tag', () => {
    metadata.tags = ['Tech-Preview', 'Debian', 'Go'];
    renderComponent(metadata);

    const badge = screen.queryAllByTestId('card-badge');
    expect(badge.length).toEqual(1);
    expect(screen.queryByText('Tech-Preview')).toBeTruthy();
  });

  it('should handle card click', async () => {
    renderComponent(metadata);

    const card = screen.getByRole('article');
    await userEvent.click(card);

    expect(onCardClick).toHaveBeenCalledWith();
  });
});

function getComponent(metadata: DevfileRegistryMetadata): React.ReactElement {
  return <SampleCard key={metadata.links.self} metadata={metadata} onClick={onCardClick} />;
}
