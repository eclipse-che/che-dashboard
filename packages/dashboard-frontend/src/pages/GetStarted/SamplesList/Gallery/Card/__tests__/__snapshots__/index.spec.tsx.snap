// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Devfile Metadata Card snapshot 1`] = `
<article
  className="pf-c-card pf-m-compact pf-m-flat pf-m-selectable sample-card"
  data-ouia-component-id="OUIA-Generated-Card-1"
  data-ouia-component-type="PF4/Card"
  data-ouia-safe={true}
  data-testid="sample-card"
  id=""
  onClick={[Function]}
  tabIndex="0"
>
  <div
    className="pf-c-card__header"
  >
    <div
      className=""
    >
      <img
        alt="Go"
        aria-label="Go"
        className="pf-c-brand sampleCardIcon"
        data-testid="sample-card-icon"
        src="/images/go.svg"
      />
    </div>
    <div
      className="pf-c-card__actions"
    />
  </div>
  <div
    className="pf-c-card__header"
  >
    Go
  </div>
  <div
    className="pf-c-card__body"
  >
    Stack with Go 1.12.10
  </div>
</article>
`;
